test/unit/spec_helper.rb

```diff
@@ -0,0 +1,27 @@
+COOKBOOK_RESOLVERS = {
+  'batali' => ['Batali', 'batali/chefspec'],
+  'berkshelf' => ['Berksfile', 'chefspec/berkshelf'],
+  'librarian' => ['Cheffile', 'chefspec/librarian']
+}.freeze
+
+require 'chefspec'
+
+if ENV['COOKBOOK_RESOLVER']
+  require COOKBOOK_RESOLVERS[ENV['COOKBOOK_RESOLVER']]
+else
+  resolver_lib = COOKBOOK_RESOLVERS.values.detect do |r_file, _r_lib|
+    File.exist?(File.join(File.dirname(__FILE__), '..', '..', r_file))
+  end
+  raise 'Failed to locate valid cookbook resolver files!' unless resolver_lib
+  puts "Resolving cookbooks from #{resolver_lib.first}"
+  require resolver_lib.last
+end
+
+class Chef
+  class ResourceBuilder
+    def emit_cloned_resource_warning
```

Comment:
Hmm? Is this truly necessary? Do the cloned resources appear because of the way the specs run?